metrics/sinks/file/FileSink.cpp

```diff
@@ -0,0 +1,52 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2020 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "FileSink.hpp"
+#include "IMeasurement.hpp"
+#include <cstdio>
+
+using namespace hpccMetrics;
+
+extern "C" IMetricSink* getSinkInstance(const std::string& name, const IPropertyTree *pSettingsTree)
+{
+    IMetricSink *pSink = new FileMetricSink(name, pSettingsTree);
+    return pSink;
+}
+
+
+FileMetricSink::FileMetricSink(const std::string &name, const IPropertyTree *pSettingsTree) :
+    MetricSink(name, "file")
+{
+    pSettingsTree->getProp("filename", fileName);
+
+    //
+    // Clear the file
+    auto handle = fopen(fileName.str(), "w");
```

Comment:
I understand the file metrics collector is more for illustrative purposes than anything, but why not append if file already contains metrics from previous instances of the filemetricksink?